chatsky/utils/context_dict/ctx_dict.py

```diff
@@ -0,0 +1,243 @@
+from __future__ import annotations
+from hashlib import sha256
+import logging
+from typing import Any, Callable, Dict, Generic, List, Mapping, Optional, Sequence, Set, Tuple, Type, TypeVar, Union, overload, TYPE_CHECKING
+
+from pydantic import BaseModel, PrivateAttr, TypeAdapter, model_serializer, model_validator
+
+from .asyncronous import launch_coroutines
+from chatsky.utils.logging import collapse_num_list
+
+if TYPE_CHECKING:
+    from chatsky.context_storages.database import DBContextStorage
+
+K = TypeVar("K", bound=int)
+V = TypeVar("V")
+
+logger = logging.getLogger(__name__)
+
+
+def get_hash(string: bytes) -> bytes:
+    return sha256(string).digest()
+
+
+class ContextDict(BaseModel, Generic[K, V]):
+    _items: Dict[K, V] = PrivateAttr(default_factory=dict)
+    _hashes: Dict[K, int] = PrivateAttr(default_factory=dict)
+    _keys: Set[K] = PrivateAttr(default_factory=set)
+    _added: Set[K] = PrivateAttr(default_factory=set)
+    _removed: Set[K] = PrivateAttr(default_factory=set)
```

Comment:
I don't like that many private attributes.
Maybe split this into two classes (context dict and context dict db connector)?
The first would implement all dict methods and have a single private attribute (context dict db connector);
and the second would hold all the fields and implement some methods (load, get, set).